### ISSUE TITLE
Crash in `tgfx::ResourceCache::detachFromCurrentThread()`

### ISSUE THREAD

**Issue by Tangdixi** [Tencent/libpag#2055]
Ran into a random crash where the system throws a SIGABRT. It seems to be coming from `tgfx::ResourceCache::detachFromCurrentThread()`. Haven't been able to nail down a solid repro yet, but it's happening.

## Which Version of libpag are you using?
libpag 4.2.110

## What Platform are you on? 
iOS 15.8 (19H370)

## Expected Behavior 
No crash

## Actual Behavior
Crash

## Code Example

Crash stack trace:
```
Last Exception :
0 libsystem_kernel.dylib _pthread_kill + 8
1 libsystem_pthread.dylib _pthread_kill + 208
2 libsystem_c.dylib _abort + 124
3 libsystem_malloc.dylib _malloc_put + 0
4 libsystem_malloc.dylib _malloc_report + 60
5 libsystem_malloc.dylib _free + 284
6 libpag tgfx::ResourceCache::detachFromCurrentThread() (ResourceCache.cpp:76)
7 libpag tgfx::Device::unlock() (Device.cpp:48)
8 libpag pag::PAGSurface::draw(pag::RenderCache*, std::__1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool) (PAGSurface.cpp:201)
9 libpag pag::PAGPlayer::flushInternal(pag::BackendSemaphore*) (PAGPlayer.cpp:293)
10 libpag pag::PAGPlayer::flush() (PAGPlayer.cpp:281)
11 libpag -[PAGView flush] (PAGView.mm:418)
12 libpag _26-[PAGView updateViewAsync]_block_invoke (PAGView.mm:460)
13 Foundation _NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
14 Foundation -[NSBlockOperation main] + 100
15 Foundation _NSOPERATION_IS_INVOKING_MAIN__ + 20
16 Foundation -[NSOperation start] + 764
17 Foundation _NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 20
18 Foundation _NSOQSchedule_f + 180
19 libdispatch.dylib _dispatch_block_async_invoke2 + 104
20 libdispatch.dylib _dispatch_client_callout + 16
21 libdispatch.dylib _dispatch_continuation_pop$VARIANT$mp + 440
22 libdispatch.dylib _dispatch_async_redirect_invoke + 588
23 libdispatch.dylib _dispatch_root_queue_drain + 340
24 libdispatch.dylib _dispatch_worker_thread2 + 172
25 libsystem_pthread.dylib _pthread_wqthread + 224
```

User footprint:
```
01-02 18:16:16.535 [ WSContainerController Daily Logins, Daily Rewards ] WillAppear
01-02 18:16:16.535 [ KSWebViewController Daily Logins, Daily Rewards ] WillAppear
01-02 18:16:38.544 [ WSContainerController ] WillAppear
01-02 18:16:38.544 [ KSMailSessionDetailViewController ] WillAppear
01-02 18:16:42.413 UIApplication WillResignActive
01-02 18:16:43.138 UIApplication DidEnterBackground
01-02 20:03:35.859 UIApplication WillEnterForeground
01-02 20:03:36.157 UIApplication DidBecomeActive
01-02 20:03:38.762 [ WSContainerController ] WillAppear
01-02 20:03:38.768 [ KSHippyViewController ] WillAppear
01-02 20:03:40.055 [ WSContainerController ] WillAppear
01-02 20:03:40.079 [ KSMultiKtvViewController ] WillAppear
01-02 20:03:43.033 [ WSTakeMicConfirmViewController ] WillAppear
01-02 20:04:20.838 UIApplication WillResignActive
01-02 20:04:32.455 UIApplication DidBecomeActive
01-02 20:04:45.994 UIApplication WillResignActive
01-02 20:05:00.461 UIApplication DidEnterBackground
01-02 20:05:25.731 UIApplication WillEnterForeground
01-02 20:05:26.026 UIApplication DidBecomeActive
01-02 20:05:51.714 UIApplication WillResignActive
01-02 20:05:55.070 UIApplication DidBecomeActive
```

Just wondering if you guys are aware of this or if it's on the radar for a future fix. Any insights would be super helpful.

**Comment by domchen** [comment]
There are some known issues with the ResourceCache when multiple threads release Resources simultaneously. Although such occurrences are relatively infrequent, we are currently working on enhancing the stability in this scenario through the refactoring of the ResourceCache class. The upcoming release version is hopeful to address and resolve this issue.

**Comment by domchen** [comment]
The fix has been merged into the main branch via PR #2057. After thorough testing, we will proceed with the release of a new version. This process may takes one to two weeks.

**Comment by Tangdixi** [comment]
> The fix has been merged into the main branch via PR #2057. After thorough testing, we will proceed with the release of a new version. This process may takes one to two weeks.

Awesome! Thanks for the quick reply on this! Looking forward to the next release!
🎉🎉🎉

**Comment by domchen** [comment]
@Tangdixi Version 4.3.47 has been released, which you can find here: https://github.com/Tencent/libpag/releases/tag/v4.3.47.
